src/core/cigar.rs

```diff
@@ -29,64 +42,71 @@ fn validate_code(code: &str) -> Result<()> {
 impl Cigar {
     pub fn new_with_code_and_raw(verfer: &Verfer, code: &str, raw: &[u8]) -> Result<Cigar> {
         validate_code(code)?;
-        Ok(Cigar { matter: Matter::new_with_code_and_raw(code, raw)?, verfer: verfer.clone() })
+        let mut cigar: Cigar = Matter::new_with_code_and_raw(code, raw)?;
```

Comment:
I believe this could also be:

```rust
let mut cigar = <Cigar as Matter>::new_with_code_and_raw(code, raw)?;
```

but I liked the syntax of the way it is written in the PR. Open to other opinions.

Reply:
I guess the as is more explicit, but I also dislike the <> so... I'm fine to leave it